src/libaktualizr/primary/aktualizr_test.cc

```diff
@@ -895,6 +895,53 @@ TEST(Aktualizr, InstallWithUpdates) {
   }
 }
 
+/**
+ * Verifies reporting of update download progress
+ *
+ * Checks actions:
+ *
+ * - [x] Report download progress
+ */
+TEST(Aktualizr, ReportDownloadProgress) {
+  // The test initialization part is repeated in many tests so maybe it makes sense
+  // to define a fixture and move this common initialization part into the fixture's SetUp() method
+  TemporaryDirectory temp_dir;
+  auto http = std::make_shared<HttpFake>(temp_dir.Path(), "hasupdates");
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, http->tls_server);
+  auto storage = INvStorage::newStorage(conf.storage);
+  Aktualizr aktualizr(conf, storage, http);
+
+  /* perhaps should be static const member of DownloadProgressReport */
```

Comment:
Static? That would mean that all instances of the class would have the same progress number, which is not desirable.

Reply:
Yes, it's a number which represents that download was successful, so clients know criterion of download success.

Reply:
Oh sorry, I misunderstood what you meant! Yes, you are probably correct. Feel free to fix it properly.